components/button/button-icon.js

```diff
@@ -35,13 +36,13 @@ class ButtonIcon extends ThemeMixin(ButtonMixin(VisibleOnAncestorMixin(RtlMixin(
 			 * REQUIRED: Preset icon key (e.g. "tier1:gear")
 			 * @type {string}
 			 */
-			icon: { type: String, reflect: true },
+			icon: { type: String, reflect: true, required: true },
```

Comment:
Also making `icon` required. The `render()` code does currently handle a missing icon gracefully by rendering `nothing`... so there could be cases where consumers were knowingly not providing an icon that will now see errors. Not sure how I feel about that, since in the docs it's "REQUIRED" and a button-icon without text visually renders nothing.